tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -139,25 +160,98 @@ public function assertLocalizedProperty(string $productReference, string $fieldN
+                    $langIso
+                ));
+            }
 
-            if ($expectedValue !== $actualValue) {
-                $langIso = Language::getIsoById($langId);
+            $actualValue = $actualValues[$langId];
 
+            if ($expectedValue !== $actualValue) {
                 throw new RuntimeException(
                     sprintf(
                         'Expected %s in "%s" language was "%s", but got "%s"',
                         $fieldName,
                         $langIso,
-                        $expectedValue,
-                        $actualValue
+                        var_export($expectedValue, true),
+                        var_export($actualValue, true)
                     )
                 );
             }
         }
     }
 
+    /**
+     * Product tags differs from other localized properties, because each locale can have an array of tags
+     * (whereas common property will have one value per language)
+     * This is why it needs some additional parsing
+     *
+     * @param array $localizedTagStrings key value pairs where key is language id and value is string representation of array separated by comma
+     *                                   e.g. [1 => 'hello, goodbye', 2 => 'bonjour,Au revoir']
+     * @param ProductForEditing $productForEditing
+     */
+    private function assertLocalizedTags(array $localizedTagStrings, ProductForEditing $productForEditing)
+    {
+        $fieldName = 'tags';
+        $actualLocalizedTags = $this->extractValueFromProductForEditing($productForEditing, $fieldName);
+
+        foreach ($localizedTagStrings as $langId => $tagsString) {
+            $langIso = Language::getIsoById($langId);
+
+            if (empty($tagsString)) {
+                // if tags string is empty, then we should not have any actual value in this language
+                if (isset($actualLocalizedTags[$langId])) {
+                    throw new RuntimeException(sprintf(
+                        'Expected no tags in %s language, but got "%s"',
+                        $langIso,
+                        var_export($actualLocalizedTags[$langId], true))
+                    );
+                }
+
+                // if above code passed it means tags in this lang is empty as expected and we can continue
+                continue;
+            }
+
+            // convert filled tags to array
+            $expectedTags = explode(',', $tagsString);
+
+            // All empty values have ben filtered out above, so if this is empty it means it is invalid
+            if (!isset($actualLocalizedTags[$langId])) {
+                throw new RuntimeException(sprintf(
+                    'Expected localized tags value "%s" is not set in %s language',
+                    var_export($expectedTags, true),
+                    $langIso
+                ));
+            }
+
+            // finally assert if both non-empty values are equal
+            Assert::assertEquals(
```

Comment:
What if they are not in the same order? I guess it shouldn't happen?

Reply:
well i hope It doesn't happen, but even if it does, we'll have a failing test and then we'll fix it, I don't think its worth investigating it now :thinking: